test/wallet-test.js

```diff
@@ -2279,4 +2299,134 @@ describe('Wallet', function() {
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [tx]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+      assert.equal(json.state, 'LOCKED');
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 1);
+      assert.equal(bal.coin, 1);
+      assert.equal(bal.unconfirmed, lockup);
+      assert.equal(bal.confirmed, lockup);
+      assert.equal(bal.ulocked, lockup);
+      assert.equal(bal.clocked, lockup);
+    });
+
+    it('should close the auction', async () => {
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      const {blocksUntilClosed} = json.stats;
+
+      for (let i = 0; i < blocksUntilClosed; i++) {
+        const entry = nextEntry(wdb);
+        await wdb.addBlock(entry, []);
+      }
+
+      {
+        const ns = await wallet.getNameState(nameHash);
+        const json = ns.getJSON(wdb.state.height, network);
+        assert.equal(json.name, 'cloudflare');
+        assert.equal(json.state, 'CLOSED');
+      }
+    });
+
+    it('should send an update for cloudflare', async () => {
+      const records = Resource.fromJSON({
+        records: [{type: 'NS', ns: 'ns1.easyhandshake.com.'}]
+      });
+
+      update = await wallet.sendUpdate('cloudflare', records);
+
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [update]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+
+      const resource = Resource.decode(ns.data);
+      assert.deepEqual(records.toJSON(), resource.toJSON());
+
+      // The unconfirmed and confirmed values should
+      // take into account the transaction fee. Assert
+      // against the value of the newly created output.
+      const val = update.output(1).value;
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 2);
+      assert.equal(bal.coin, 2);
+      assert.equal(bal.unconfirmed, val);
+      assert.equal(bal.confirmed, val);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
+    });
+
+    it('should remove a block and update balances correctly', async () => {
+      const cur = curEntry(wdb);
+      await wdb.removeBlock(cur);
+
+      const bal = await wallet.getBalance();
+
+      const val = update.output(1).value;
+      assert.equal(bal.tx, 2);
+      assert.equal(bal.coin, 2);
+
+      // The unconfirmed balance includes value in the mempool
+      // and the chain itself. The reorg'd tx can be included
+      // in another block so the unconfirmed total does not
+      // include the tx fee, since that value has been effectively
+      // spent already.
+      assert.equal(bal.unconfirmed, val);
+      assert.equal(bal.confirmed, lockup);
+      assert.equal(bal.ulocked, 0);
```

Comment:
`bal.ulocked` is `0`, because the transaction that unlocks the `lockup` value is still in the wallet. However, it is unconfirmed.